### PR TITLE
Add VAST 4.2 support

### PR DESCRIPTION
**Pull request by myslak71**

### CODE REVIEW

File: testdata/vast4.2_click_through_in_wrapper.xml [hunk 1]
```diff
@@ -0,0 +1,90 @@
+<VAST version="4.0"
```
Comment (pabramowicz) [inline testdata/vast4.2_click_through_in_wrapper.xml]: Why all versions are set to 4.0 and not 4.2?

**Review by pabramowicz — APPROVED** [review]

File: testdata/vast4.2_click_through_in_wrapper.xml (hunk 1 above)
Comment (myslak71) [reply testdata/vast4.2_click_through_in_wrapper.xml]: will fix

File: testdata/vast4_universal_ad_id.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<VAST version="4.0" 
-  xmlns:xs="http://www.w3.org/2001/XMLSchema" 
+<VAST version="4.2"
```
Comment (fizyk) [inline testdata/vast4_universal_ad_id.xml]: shouldn't we rather copy and duplicate the testdata for 4.2? Or does it really needs changing?

File: testdata/vast_adaptv_attempt_attr.xml [hunk 3]
```diff
@@ -101,11 +101,21 @@
...
-						<Icon program="DAA" width="77" height="15" xPosition="right" yPosition="top" offset="00:00:00">
+						<Icon program="DAA" width="77" height="15" xPosition="right" yPosition="top" offset="00:00:00" altText="spekemat" hoverText="kick it!">
 							<StaticResource creativeType="image/png"><![CDATA[https://s.aolcdn.com/ads/adchoices.png]]></StaticResource>
 
 							<IconClicks>
 								<IconClickThrough><![CDATA[https://adinfo.aol.com]]></IconClickThrough>
+								<IconClickFallbackImages>
+									<IconClickFallbackImage width="69" height="69">
+										<AltText><![CDATA[https://myslaki.pl]]></AltText>
+										<StaticResource creativeType="image/png"><![CDATA[https://myslaki.pl/fallback_image.png]]></StaticResource>
```
Comment (fizyk) [inline testdata/vast_adaptv_attempt_attr.xml]: use .test domain tlds

File: testdata/vast4.2_click_through_in_wrapper.xml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                            </MediaFile>
+                            <MediaFile id="5246" delivery="progressive" type="video/mp4" bitrate="600" width="640"
+                                       height="360" minBitrate="500" maxBitrate="700" scalable="1"
+                                       maintainAspectRatio="1" codec="0">
+                                <![CDATA[https://iabtechlab.com/wp-content/uploads/2017/12/VAST-4.0-Short-Intro-low-resolution.mp4]]>
+                            </MediaFile>
+                        </MediaFiles>
+                        <VideoClicks>
+                            <ClickThrough id="blog">
+                                <![CDATA[https://iabtechlab.com]]>
```
Comment (fizyk) [inline testdata/vast4.2_click_through_in_wrapper.xml:53]: use .test tlds

**Review by fizyk — CHANGES_REQUESTED** [review]

File: testdata/vast4_universal_ad_id.xml (hunk 2 above)
Comment (myslak71) [reply testdata/vast4_universal_ad_id.xml]: hmm I think we can leave it as it is (4.0) in this test data but for the other two datasets I'd leave 4.2 cause they contain features introduced by 4.2

File: testdata/vast4.2_click_through_in_wrapper.xml (hunk 4 above)
Comment (jotes) [reply testdata/vast4.2_click_through_in_wrapper.xml:53]: if the tests doesn't need to validate the data/responses from the endpoints, it's better to not use domains of third parties.

**Review by luszak — APPROVED** [review]

File: testdata/vast4_universal_ad_id.xml [hunk 5]
```diff
@@ -43,7 +43,7 @@
             </MediaFiles>
             <VideoClicks>
               <ClickThrough id="blog">
-                <![CDATA[https://iabtechlab.com]]>
+                <![CDATA[https://test.com]]>
```
Comment (fizyk) [inline testdata/vast4_universal_ad_id.xml]: it's still .com tld, please use .test tld

**Review by fizyk — CHANGES_REQUESTED** [review]
https://en.wikipedia.org/wiki/Top-level_domain

File: testdata/vast4_universal_ad_id.xml (hunk 5 above)
Comment (myslak71) [reply testdata/vast4_universal_ad_id.xml]: sure!

**Review by fizyk — APPROVED** [review]

**Pull request merged**